### PR TITLE
fix: mint client state machine migration

### PR DESCRIPTION
**Pull request by bradleystachurski**
Fixes https://github.com/fedimint/fedimint/issues/6338

This only fixes one failure of upgrade tests. Since `master` and `releases/v0.5` are still broken after including this fix, I haven't kicked off a run in CI, only local tests. To verify locally against these changes, first checkout the commit that introduced this failure and run upgrade tests for `fedimint-cli`

```
git checkout 8d156fd39049203fc134c2debe99e1182c3a6d60
TEST_KINDS=fedimint-cli just test-upgrades v0.4.0 current &> results.log
```

Expected failure message

```
Error: command: fedimint-cli --data-dir=/tmp/nix-shell.lh5Zqz/devimint-2050963-520/clients/reissue-client-0 deposit-address

Caused by:
    exit status: 1
    stdout:
    {
      "error": "Decoding named block MintOOBStateMachineV1 {} state"
    }
```

After applying the changes in this PR, `fedimint-cli` upgrade tests pass.

It's worth noting that I verified each variant of `MintClientStateMachines` that's included in this migration (`Output`, `Input`, and `OOB`) failed individually prior to adding the logic to parse the struct length, so I'm confident all paths are tested in the upgrade tests.

### CODE REVIEW

File: modules/fedimint-mint-client/src/client_db.rs [hunk 1]
```diff
@@ -128,7 +128,9 @@ pub(crate) fn migrate_state_to_v2(
 
     let new_mint_state_machine = match mint_client_state_machine_variant {
         0 => {
-            let old_state = MintOutputStateMachineV1::consensus_decode(cursor, &decoders)?;
+            let _output_sm_len = u16::consensus_decode(cursor, &decoders)?;
```
Comment (bradleystachurski) [inline modules/fedimint-mint-client/src/client_db.rs:131]: We're failing to decode the structs in the migration if we don't first read the length.

Comment (dpc) [reply modules/fedimint-mint-client/src/client_db.rs:131]: Makes sense. I always forget about this extra len in variants.

But, I copied the code here from `modules/fedimint-dummy-client/src/db.rs` `get_v1_migrated_state` and I don't see it decoding any len either, so does it mean it's broken as well, and just was never actually tested?

Comment (m1sterc001guy) [reply modules/fedimint-mint-client/src/client_db.rs:131]: Hmm we do test it in `test_client_db_migrations` of the dummy tests crate, but perhaps its decoding to the wrong values. We don't test that part. I'll check

Comment (bradleystachurski) [reply modules/fedimint-mint-client/src/client_db.rs:131]: We can address the dummy client migrations in a followup. Right now this is making it difficult to continue triaging the rest of the failures in the upgrade tests, so if there isn't a logical error let's land soon :slightly_smiling_face:

**Review by m1sterc001guy — APPROVED** [review]

File: modules/fedimint-mint-client/src/client_db.rs (hunk 1 above)
Comment (m1sterc001guy) [reply modules/fedimint-mint-client/src/client_db.rs:131]: All the reason why the dummy migration tests work is just a happy accident lol. The `Unreachable` state decodes a `u16` and I guess originally I thought this was the `ModuleInstanceId`, but its actually the length. 

```
impl Decodable for Unreachable {
    fn consensus_decode<R: std::io::Read>(
        reader: &mut R,
        modules: &ModuleDecoderRegistry,
    ) -> Result<Self, fedimint_core::encoding::DecodeError> {
        let module_instance_id = ModuleInstanceId::consensus_decode(reader, modules)?;
        let operation_id = OperationId::consensus_decode(reader, modules)?;
        let txid = TransactionId::consensus_decode(reader, modules)?;
        let amount = Amount::consensus_decode(reader, modules)?;

        Ok(Unreachable {
            _module_instance_id: module_instance_id,
            operation_id,
            txid,
            amount,
        })
    }
}
```

I can submit a quick fix.

Comment (m1sterc001guy) [reply modules/fedimint-mint-client/src/client_db.rs:131]: https://github.com/fedimint/fedimint/pull/6362

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.5`:
- #6363

**Pull request merged**